server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageRepositoryCommand.java

```diff
@@ -50,16 +53,29 @@ public void update(CruiseConfig modifiedConfig) {
         PackageRepositories repositories = modifiedConfig.getPackageRepositories();
         repositories.replace(oldRepo, newRepo);
         modifiedConfig.setPackageRepositories(repositories);
+
+        updatePackageRepositoryConfigurationOnAssociatedPipelines(modifiedConfig);
     }
 
     @Override
     public boolean canContinue(CruiseConfig cruiseConfig) {
         return super.canContinue(cruiseConfig) && isIdSame() && isRequestFresh();
     }
 
+    private void updatePackageRepositoryConfigurationOnAssociatedPipelines(CruiseConfig modifiedConfig) {
+        List<PipelineConfig> pipelinesWithPackageRepo = modifiedConfig.pipelinesAssociatedWithPackageRepository(newRepo);
+        pipelinesWithPackageRepo.forEach(pipelineConfig -> {
+            pipelineConfig.packageMaterialConfigs().forEach(packageMaterialConfig -> {
+                if (packageMaterialConfig.getPackageDefinition().getRepository().getId().equals(newRepo.getId())) {
```

Comment:
Same here: the check has performed in `modifiedConfig.pipelinesAssociatedWithPackageRepository`